hash_practice/exercises.py

```diff
@@ -1,19 +1,48 @@
+from collections import defaultdict
 
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n*m) because m depends on how long the string is, so that is variable 

```

Comment:
Very good thought to consider `m` in this case, I would probably have included it too. However two things:

1. Because you call `sort` on the word, that is going to take O(m*log(m)) time so that has to be accounted for too if you're going to account `m`. However...
2. We are accepting O(n) for this because while `m` definitely has an effect, with the assumption that the input words are English words, the number of letters in them are pretty limited. On average there are only about 5 letters per word. However, the size of the list of words is unbounded, so it ends up being the dominating factor to the point we can just do O(n).